HW_8/task 8.2.js

```diff
@@ -0,0 +1,269 @@
+  if (foundEnterprise == 0) throw new Error('Предприятия с данным id не существует');
+  foundEnterprise[0].name = newName;
+  
+  return JSON.stringify(enterprises);
+}
+
+console.log(editEnterprise(1, 'ООО "Рога и копыта"'));
+
+// 6. Написать функцию для редактирования названия отдела. Принимает в качестве аргумента id отдела и новое имя отдела.
+
+// Пример:
+// editDepartment(7, "Новое название отдела")
+
+// Функция-помощник для получения отдела по id отдела
+const getDepartmentById = (id, []) => {
+  let result;
+  enterprises.forEach((enterprise) => enterprise.departments.forEach((department) => {
+      if (department.id === id) {
+        result = department;
+      }
+    })
+  );
+  
+  return result;
+};
+
+const editDepartment = (id, newName) => {
+  const foundDepartment = getDepartmentById(id, enterprises);
+  if (foundDepartment == undefined) throw new Error('Отдела с данным id не существует');
+  foundDepartment.name = newName;
+  
+  return JSON.stringify(enterprises);
+}
+
+console.log(editDepartment(7, "Новое название отдела"));
+
+// 7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+
+// Пример:
+// deleteEnterprise(1)
+
+const deleteEnterprise = (id) => {
+  const foundEnterprise = getEnterpriseById(id, enterprises);
+  if (foundEnterprise.length === 0) throw new Error('Предприятия с данным id не существует');
+  
+  const findIndex = enterprises.indexOf(...foundEnterprise);
+  enterprises.splice(findIndex, 1)
+  
+  return enterprises;
+}
+
+console.log(deleteEnterprise(1));
+
+// 8. Написать функцию для удаления отдела. В качестве аргумента принимает id отдела. Удалить отдел можно только, если в нем нет сотрудников.
+
+// Пример:
+// deleteDepartment(3)
+
+const deleteDepartment = (id) => {
+  if (typeof id != 'number') throw new Error('Введите верные данные');
+  const foundEnterprise = getEnterpriseByDepartment(id);
+  
+  if (foundEnterprise.length === 0) throw new Error('Отдела с данным id не существует');
+  const findObject = foundEnterprise[0].departments.filter(department => department.id === id);
+  
+  if (findObject[0].employees_count !== 0) throw new Error('Нельзя удалить отдел с сотрудниками');
+  const findIndexOfDepartment = foundEnterprise[0].departments.indexOf(...findObject);
+  foundEnterprise[0].departments.splice(findIndexOfDepartment, 1);
+  
+  return JSON.stringify(enterprises);
+}
+
+console.log(deleteDepartment(3));
+console.log(deleteDepartment(10));
+
+// 9. Написать функцию для переноса сотрудников между отделами одного предприятия. В качестве аргумента принимает два значения: id отдела, из которого будут переноситься сотрудники и id отдела, в который будут переноситься сотрудники).
+
+// Пример:
+// moveEmployees(2, 3)
+
+const moveEmployees = (id1, id2) => {
+  const getFromDepartment = getDepartmentById(id1, enterprises);
+  const addToDepartment = getDepartmentById(id2, enterprises);
+  
```

Comment:
No check that this departments are nested in one enterprise

Reply:
I forgot to add this check, will fix in the next commit